sdk/core/Azure.Core/samples/Events.md

```diff
@@ -0,0 +1,191 @@
+  operations that take a token so cancellation (via something like
+  `new CancellationTokenSource(TimeSpan.FromSeconds(10)).Token`, for example)
+  will correctly propagate.
+
+- There is a `SyncAsyncEventArgs.RunSynchronously` flag indicating whether your
+  handler was invoked synchronously or asynchronously.  In general,
+
+    - If you're calling sync methods on your client, you should use sync methods
+      to implement your event handler.  You can return `Task.CompletedTask`.
+    - If you're calling async methods on your client, you should use async
+      methods where possible to implement your event handler.
+    - If you're not in control of how the client will be used or want to write
+      safer code, you should check the `RunSynchronously` property and call
+      either sync or async methods as directed.
+
+  There are code examples of all three situations below to compare.  Please also
+  see the note at the very end discussing the dangers of sync-over-async to
+  understand the risks of not using the `RunSynchronously` flag.
+
+- Most events will customize the event data by deriving from `SyncAsyncEventArgs`
+  and including details about what triggered the event or providing options to
+  react.  Many times this will include a reference to the client that raised the
+  event in case you need it for additional processing.
+
+When an event using `SyncAsyncEventHandler` is raised, the handlers will be
+executed sequentially to avoid introducing any unintended parallelism.  The
+event handlers will finish before returning control to the code path raising the
+event.  This means blocking for events raised synchronously and waiting for the
+returned `Task` to complete for events raised asynchronously.  Any exceptions
+thrown from a handler will be wrapped in a single `AggregateException`.  Finally,
+a [distributed tracing span](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Diagnostics.md#distributed-tracing)
+is wrapped around your handlers using the event name so you can see how long
+your handlers took to run, whether they made other calls to Azure services, and
+details about any exceptions that were thrown.
+
+The rest of the code samples are using a fictitious `AlarmClient` to demonstrate
+how to handle `SyncAsyncEventHandler` events.  There are `Snooze` and
+`SnoozeAsync` methods that both raise a `Ring` event.
+
+## Adding a synchronous event handler
+
+If you're using the synchronous, blocking methods of a client (i.e., methods
+without an `Async` suffix), they will raise events that require handlers to
+execute synchronously as well.  Even though the signature of your handler
+returns a `Task`, you should write regular sync code that blocks and return
+`Task.CompletedTask` when finished.
+
+```C# Snippet:Azure_Core_Samples_EventSamples_SyncHandler
+var client = new AlarmClient();
+client.Ring += (SyncAsyncEventArgs e) =>
```

Comment:
Need an example with unsubscribing from event.

Reply:
I don't think there's a good reason to.  There's nothing special about our model that changes unsubscribing and it's not a thing we'd expect most customers to do.